File: Leetcode/python/2220-min-bit-flip.py
class Solution:
    def minBitFlips(self, start: int, goal: int) -> int:
        startNum = bin(start)[2:]
        endNum = bin(goal)[2:]
        maxLen = max(len(startNum), len(endNum))

        startNum = startNum.zfill(maxLen)
        endNum = endNum.zfill(maxLen)

        res = 0
        for s, e in zip(startNum, endNum):
            if s != e:
                res += 1

        return res

File: Leetcode/typescript/2220-min-bit-flip.ts
function minBitFlips(start: number, goal: number): number {
  let startNum = start.toString(2);
  let endNum = goal.toString(2);
  const maxLen = Math.max(startNum.length, endNum.length);

  startNum = startNum.padStart(maxLen, "0");
  endNum = endNum.padStart(maxLen, "0");

  let res = 0;

  for (let i = 0; i < maxLen; i++) {
    if (startNum[i] !== endNum[i]) {
      res++;
    }
  }
  return res;
}
